projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -878,6 +897,10 @@ export class IgxDatePickerComponent implements IDatePicker, ControlValueAccessor
             this.attachTemplateBlur();
          });
         this.attachTemplateBlur();
+
```

Comment:
this should be moved in `ngAfterViewChecked`. If you change the `mode` at run time `aria-labeledby` is lost.